docs/rendering_templates.md

```diff
@@ -94,16 +94,38 @@ Given the information in the **BOSH Job Spec Secrets** and the **Desired Manifes
 - a list of all **BOSH Variables** and their corresponding `ExtendedSecrets`
 - all **BOSH Properties** - both from specs and the **Desired Manifest**
 - an understanding of what type of template is being rendered (binary or not)
+- the number of replicas for each instance group
+- the number of availability zones for each instance group
 
-### Service Addresses
+### Addresses
 
-Service addresses are calculated in the following manner:
+Addresses for individual pods are calculated in the following manner:
 
 ```
-<JOB_NAME>-<INSTANCE_GROUP_NAME>-<DEPLOYMENT_NAME>.<KUBE_NAMESPACE>.<KUBE_SERVICE_DOMAIN>
+<INSTANCE_GROUP_NAME>-<INDEX>-<DEPLOYMENT_NAME>.<KUBE_NAMESPACE>.<KUBE_SERVICE_DOMAIN>
 ```
 
-> E.g.: `cloud-controller-ng-api-cfdeployment.mycf.svc.cluster.local`
+> E.g.: `api-group-0-cfdeployment.mycf.svc.cluster.local`
+
+The index is calculated as the index of the pod, multiplied by the AZ index of the pod.
+A **Kubernetes Service** is created for each pod of a StatefulSet, that has a selector for the correct pod name(s).
+e.g.
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: api-group-0-cfdeployment
+spec:
+  type: ClusterIP
+  selector:
+    statefulset.kubernetes.io/pod-name: api-group-v1-0
+    statefulset.kubernetes.io/pod-name: api-group-v2-0
```

Comment:
Not sure where the `v1` and `v2` come from.

I assume we have separate ExtendedStatefulSets for each AZ, so I would like to see the AZ name in the pod name as well as in the service name.

Reply:
This is how `ExtendedStatefulSets` work - `v1` and `v2` are revisions. See [here](https://github.com/cloudfoundry-incubator/cf-operator/blob/master/docs/controllers/extendedstatefulset.md#extended-upgrade-support) for more details.

About AZs - `ExtendedStatefulSets` are AZ-aware - so we'll have a separate `StatefulSet` for each AZ.